init-interfaces.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -ex
+
+is_con_exists() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+is_con_active() {
+	local con_name=$1
+	if [[ $(nmcli -t -f NAME con show --active | grep -w "$con_name") ]]; then
+		echo "true"
+	else
+		echo "false"
+	fi
+}
+
+
+if [[ ! -f /boot/mac_addresses ]] ; then
+  echo "no mac address configuration file found .. exiting"
+  exit 1
+fi
+
+primary_mac=$(cat /boot/mac_addresses | awk -F= '/PRIMARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+secondary_mac=$(cat /boot/mac_addresses | awk -F= '/SECONDARY_MAC/ {print $2}' | tr '[A-Z]' '[a-z]')
+
+default_device=""
+secondary_device=""
+default_connection_name=""
+secondary_connection_name=""
+
+for dev in $(nmcli device status | awk '/ethernet/ {print $1}'); do
+  dev_mac=$(nmcli -g GENERAL.HWADDR dev show $dev | sed -e 's/\\//g' | tr '[A-Z]' '[a-z]')
+  case $dev_mac in
+    $primary_mac)
+      default_device=$dev
+      default_connection_name=$(nmcli -g GENERAL.CONNECTION dev show $dev)
```

Comment:
Could we use the UUID instead so we survive presence of connections with duplicates names?